File: 0394-decode-string/0394-decode-string.cpp
class Solution {
private:
    string dfs(int& idx, string s, int t){
        string temp = "";
        while(idx<s.length() && s[idx]!=']'){
            if(isdigit(s[idx])){
                t = t*10+(s[idx]-'0');
                idx+=1;
            } else if(isalpha(s[idx])){
                temp.push_back(s[idx]);
                idx+=1;
            } else if(s[idx] == '['){
                idx+=1;
                int f = t;
                string res = dfs(idx, s, 0);
                
                while(f--){
                    temp+=res;
                }
                idx+=1;
                t=0;
            }
        }
        
        return temp;
    }
public:
    string decodeString(string s) {
        string answer="";
        int idx=0;
        return dfs(idx, s, 0);
        
    }
};

File: 0394-decode-string/0394-decode-string.ts
let i: number;

function dfs(s: string, t: number){
    let temp: string = "";
    
    while(i<s.length && s[i]!="]"){
        if(!isNaN(parseInt(s[i]))){
               t=t*10 + (s[i].charCodeAt(0) - "0".charCodeAt(0));
        } else if(s[i] == "["){
            i+=1;
            let f = t;
            let res = dfs(s, 0);
            let ans="";
            while(f--){
                ans+=res;
            }
            temp+=ans;
            t=0;
        } else{
            temp+=s[i];
        }
        i+=1;
    }
    
    return temp;
}


function decodeString(s: string): string {
    i=0;
    let t=0;
    return dfs(s, t);
};